jsettlers.mapcreator/src/jsettlers/mapcreator/control/EditorControl.java

```diff
@@ -0,0 +1,861 @@
+import jsettlers.mapcreator.stat.StatisticsDialog;
+import jsettlers.mapcreator.tools.SetStartpointTool;
+import jsettlers.mapcreator.tools.Tool;
+import jsettlers.mapcreator.tools.landscape.ResourceTool;
+import jsettlers.mapcreator.tools.objects.PlaceBuildingTool;
+import jsettlers.mapcreator.tools.shapes.ShapeType;
+
+/**
+ * Controller for map editing
+ * 
+ * @author Andreas Butti
+ */
+public class EditorControl implements IMapInterfaceListener, ActionFireable, IPlayerSetter, IScrollToAble {
+
+	/**
+	 * Map data
+	 */
+	private MapData data;
+
+	/**
+	 * Map drawing
+	 */
+	private MapGraphics map;
+
+	/**
+	 * Currently active tool
+	 */
+	private Tool tool = null;
+
+	/**
+	 * Currently selected player
+	 */
+	private int currentPlayer = 0;
+
+	/**
+	 * Undo / Redo stack
+	 */
+	private UndoRedoHandler undoRedo;
+
+	/**
+	 * To scrol to positions
+	 */
+	private MapInterfaceConnector connector;
+
+	/**
+	 * Header of the current open map
+	 */
+	private MapFileHeader header;
+
+	/**
+	 * Window displayed
+	 */
+	private EditorFrame window;
+
+	/**
+	 * Open GL Contents (Drawing)
+	 */
+	private MapContent mapContent;
+
+	/**
+	 * Sidebar with the tools
+	 */
+	private ToolSidebar toolSidebar = new ToolSidebar(this) {
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		protected void changeTool(Tool lastPathComponent) {
+			EditorControl.this.changeTool(lastPathComponent);
+		}
+
+	};
+
+	/**
+	 * Sidebar with all tabs
+	 */
+	private Sidebar sidebar = new Sidebar(toolSidebar, this);
+
+	/**
+	 * Validates the map for errors
+	 */
+	private MapValidator validator = new MapValidator();
+
+	/**
+	 * Timer for redrawing
+	 */
+	private final Timer redrawTimer = new Timer(true);
+
+	/**
+	 * Action to fix all errors automatically, if clear what to do
+	 */
+	private AutoFixErrorAction autoFixErrorAction;
+
+	/**
+	 * Always display resources
+	 */
+	private boolean showResourcesAlways = false;
+
+	/**
+	 * Show the resources because the current tool requests it
+	 */
+	private boolean showResourcesBecauseOfTool = false;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param header
+	 *            Header of the file to open
+	 * @param ground
+	 *            Ground to use for the new map
+	 */
+	public EditorControl(MapFileHeader header, ELandscapeType ground) {
+		init(header, new MapData(header.getWidth(), header.getHeight(), header.getMaxPlayer(), ground));
+	}
+
+	/**
+	 * Constructor
+	 * 
+	 * @param loader
+	 *            Map to load
+	 * @throws MapLoadException
+	 */
+	public EditorControl(MapLoader loader) throws MapLoadException {
+		MapData data = new MapData(loader.getMapData());
+		MapFileHeader header = loader.getFileHeader();
+		init(header, data);
+	}
+
+	/**
+	 * Initialize editor, called from constructor
+	 * 
+	 * @param header
+	 *            Header to use
+	 * @param mapData
+	 *            Map to use
+	 */
+	private void init(MapFileHeader header, MapData mapData) {
```

Comment:
instead of an init method, you can make a private constructor and let the other constructors call this constructor. This can also help making more fields final.